src/base/logger.cpp

```diff
@@ -89,9 +91,39 @@ void Logger::addPeer(const QString &ip, const bool blocked, const QString &reaso
     const Log::Peer temp = {m_peerCounter++, QDateTime::currentMSecsSinceEpoch(), ip.toHtmlEscaped(), blocked, reason.toHtmlEscaped()};
     m_peers.push_back(temp);
 
+    locker.unlock();
+
     emit newLogPeer(temp);
 }
 
+const Log::Msg& Logger::getMessage(const int id) const
+{
+    QReadLocker locker(&m_lock);
+
+    const int start = m_msgCounter - m_messages.size();
+    const int end = m_msgCounter - 1;
+    const bool inRange = ((start <= id) && (id <= end));
+
+    if (!inRange)
+        return {-1, -1, Log::CRITICAL, ""};
+
+    return m_messages.at(id - start);
```

Comment:
`at()` method will do boundary checks and throw exception if not valid, since you already validates the index why not use `operator[]`?